Makefile.source.mjs

```diff
@@ -0,0 +1,359 @@
+import "shelljs/make.js";
+import path from "path";
+import { execFileSync } from "child_process";
+import { writeFileSync } from "fs";
+
+/**
+ * @type {import("shelljs")}
+ */
+const shell = global;
+const target = global.target;
+
+const SOURCES = ["packages", "codemods", "eslint"];
+
+const YARN_PATH = shell.which("yarn").stdout;
+const NODE_PATH = process.execPath; // `yarn node` is so slow on Windows
+
+shell.config.verbose = true;
+
+function print(msg) {
+  console.log(msg);
+}
+
+function exec(executable, args, cwd, inheritStdio = true) {
+  print(
+    `${(executable.replaceAll(YARN_PATH), "yarn").replaceAll(
+      NODE_PATH,
+      "node"
+    )} ${args.join(" ")}`
+  );
+
+  try {
+    return execFileSync(executable, args, {
+      stdio: inheritStdio ? "inherit" : undefined,
+      cwd: cwd && path.resolve(cwd),
+      env: process.env,
+    });
+  } catch (error) {
+    if (inheritStdio && error.status != 0) {
+      console.error(
+        new Error(
+          `\ncommand: ${executable} ${args.join(" ")}\ncode: ${error.status}`
+        )
+      );
+      // eslint-disable-next-line no-process-exit
+      process.exit(error.status);
+    }
+    throw error;
+  }
+}
+
+function yarn(args, cwd, inheritStdio) {
+  return exec(YARN_PATH, args, cwd, inheritStdio);
+}
+
+function node(args, cwd, inheritStdio) {
+  return exec(NODE_PATH, args, cwd, inheritStdio);
+}
+
+function env(fun, env) {
+  const envBak = process.env;
+  process.env = { ...envBak, ...env };
+  fun();
+  process.env = envBak;
+}
+
+/**
+ * CLEAN
+ */
+
+target["clean-all"] = function () {
```

Comment:
I fear that the tasks in `Makefile` and the tasks in `Makefile.source.js` fill easily get out of sync. Can we update `Makefile` to delegate to `Makefile.source.js`? for example,
```js
clean-all:
  node ./Makefile.js clean-all
```

Reply:
I can also migrate all the required steps into makejs if you want, what do you think? (Only some publishing operations related are not migrated now)